inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -128,6 +129,13 @@ void MKLDNNPlugin::MKLDNNInferRequest::PushInputData() {
             default:
                 THROW_IE_EXCEPTION << "Unsupported input precision " << input.second->getTensorDesc().getPrecision();
         }
+
+        // User can initialize input via setBlob API using tensorDesc with default (ANY) layout.
+        // Currently IE doesn't specify behavior in such scenario, so we assume real layout is equal to the network input.
+        if (input.second->getTensorDesc().getLayout() == InferenceEngine::ANY) {
+            input.second->getTensorDesc().setLayout(_networkInputs[input.first]->getLayout());
+        }
```

Comment:
can we add a behavior test for it?

Reply:
@mandrono will do in separate PR in bounds of bigger test coverage activity for inputs/outputs handling: 50548